versions.props

```diff
@@ -0,0 +1,19 @@
+org.slf4j:slf4j-api = 1.7.5
+com.google.guava:guava = 28.0-jre
+org.apache.avro:avro = 1.8.2
+org.apache.hadoop:* = 2.7.3
+org.apache.hive:hive-standalone-metastore = 1.2.1
+org.apache.orc:orc-core = 1.5.5
+org.apache.parquet:parquet-avro = 1.10.0
+org.apache.spark:spark-hive_2.11 = 2.4.0
+org.apache.pig:pig = 0.14.0
+com.fasterxml.jackson.*:* = 2.7.9
```

Comment:
Why did we change Jackson version from 2.6.7 to 2.7.9? Spark 2.4.0 uses 2.6.7.1 and I am seeing the following error:

```
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Incompatible Jackson version: 2.7.9
	at com.fasterxml.jackson.module.scala.JacksonModule$class.setupModule(JacksonModule.scala:64)
	at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:19)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:730)
	at org.apache.spark.rdd.RDDOperationScope$.<init>(RDDOperationScope.scala:82)
	at org.apache.spark.rdd.RDDOperationScope$.<clinit>(RDDOperationScope.scala)
	... 81 more
```

Reply:
Sorry, I missed the discussion in this [comment](https://github.com/apache/incubator-iceberg/pull/241#discussion_r298720187)

Reply:
Do we need to force everything down to the lower version?

Reply:
Wait that doesn't make sense, in the above thread we observed that spark-sql was pulling in Jackson 2.7.9. So if spark-sql is pulling in Jackson 2.7.9 but some other Spark module -presumably spark-core - pulls in Jackson 2.6.7.1, how is Spark itself supposed to work with internally inconsistent Jackson versions?

Reply:
Checking

Reply:
If I remove Jackson from `versions.props`, I will get the following in `versions.lock`:

```
com.fasterxml.jackson.core:jackson-annotations:2.6.7 (4 constraints: ee47c3bd)
com.fasterxml.jackson.core:jackson-core:2.7.9 (4 constraints: be433a35)
com.fasterxml.jackson.core:jackson-databind:2.6.7.1 (6 constraints: 206018be)
com.fasterxml.jackson.module:jackson-module-paranamer:2.7.9 (1 constraints: e0154200)
com.fasterxml.jackson.module:jackson-module-scala_2.11:2.6.7.1 (1 constraints: 7f0da251)
```

The stacktrace above is when `com.fasterxml.jackson.module.scala.JacksonModule` calls `com.fasterxml.jackson.databind.SetupContext$getMapperVersion`. So, previously this check succeded. 

spark-core does seem to pull in `jackson-module-scala_2.11:2.6.7.1`

Reply:
I am wondering if anybody else experienced this. Maybe, my config is wrong.

Reply:
Well Spark upstream is pushing the Jackson version up to 2.9.9: https://github.com/apache/spark/blob/master/pom.xml#L173. Meanwhile we don't hit this because we don't use shadow jars instead of using mega-shadow-jars for most things. We also bumped Jackson in our fork: https://github.com/palantir/spark/blob/master/pom.xml#L182